src/relentless/optimize/objective.py

```diff
@@ -322,10 +323,13 @@ def compute(self, variables, directory=None):
             The result, which has unknown value ``None`` and known gradient.
 
         """
-        if self.thermo.rdf is None:
-            raise ValueError(
-                "EnsembleAverage needs to compute RDF, specify parameters."
-            )
+        if isinstance(self.target, Ensemble) and isinstance(
+            self.thermo, EnsembleAverage
+        ):
```

Comment:
Do you think this needs a hidden attribute to reduce duplication? It's used more than once but not a ton, so I wasn't sure.